xl2times/transforms.py

```diff
@@ -369,6 +384,21 @@ def process_flexible_import_table(
 
         attribute = "attribute"
         if table.tag != datatypes.Tag.tfm_upd:
+
+            # Check for duplicate DF columns
+            duplicated_cols = [
+                item
+                for item, count in collections.Counter(data_columns).items()
+                if count > 1
+            ]
+            if len(duplicated_cols) > 0:
+                logger.warning(
+                    f"Duplicate data columns in table: {duplicated_cols}.  Dropping first duplicated column. Table: \n{repr(table)}"
+                )
+                # drop duplicate Df columns
+                df = df.loc[:, ~df.columns.duplicated(keep="last")]
```

Comment:
What if the duplicated columns have empty entries? E.g.:

| column | column |
| --- | --- |
| 2 | |
|   |  3|
| 4 | 5| 

I don't know what Veda would do, but I would expect the resulting column to be:

| column |
| --- |
| 2 |
|  3|
| 5|

Reply:
Our resident VEDA expert told me that this matches what VEDA's behaviour, but I'll double check with him.
Is your plan to try matching VEDA, or be more restrictive and raise exceptions in these kinds of ambiguous cases?

Reply:
It is a tricky question. Ideally I would like the tool to be useful out-of-the-box; at the same it should provide feedback when the user is making an error. Maybe we could introduce something like a `--safe` mode in the future?

Reply:
I think `--safe` (or --strict?) is a great idea, as warnings are too easy to ignore, and some could have difficult to determine downstream implications.

Reply:
As far as I know, VEDA2 **does not allow** fully duplicate columns (VEDA-FE did allow).  See: https://veda-documentation.readthedocs.io/en/latest/pages/Migration.html#before-you-start
But VEDA2 still accepts e.g. `Input` and `Input~ANNUAL`, and as usual, the last value should survive (rightmost).

Reply:
Will test whether not allowing fully duplicate columns is still the case in Veda (I believe this is due to the technical limitations of the technologies uses and not design considerations).

Current behaviour of the tool is that the last value should survive whether the column is fully duplicate or not. I am inclined keeping it this way supplemented by warning for consistency reasons and simplicity.

Reply:
Hmm... I see the comment `drop duplicate Df columns` with `~df.columns.duplicated(keep="last")` in the code. Just to confirm that this does not really drop any whole column (it should not)?  If the last _duplicate_ column has a blank cell while there is a value in a preceding duplicate column on that row, you should not drop the value in that preceding column, but keep it, shouldn't you?

Reply:
Well, if accepted as is, the changes in this PR would modify the current behaviour of the tool which is "last value should survive whether the column is fully duplicate or not".

Reply:
> Well, if accepted as is, the changes in this PR would modify the current behaviour of the tool which is "last value should survive whether the column is fully duplicate or not".

This will keep the right-most column of any set of columns with identical _names_.  It doesn't consider the column _contents_.  My commenting should definitely be more explicit about this.

Here's an example of this codes' behaviour:
```python 
df = pd.DataFrame({'a': [1, 2], 'a2': [3, 4], 'b': [5, 6]})
df = df.rename(columns={'a2': 'a'})  # make a second column named 'a'
print(df)
   a  a  b
0  1  3  5
1  2  4  6

# drop columns with duplicate names
df = df.loc[:, ~df.columns.duplicated(keep="last")]
print(df)
   a  b
0  3  5
1  4  6
```
Is this what we want to match VEDA's current behaviour?

Reply:
To match VEDA's behaviour, I think one should consider the contents (blank cells vs. cells with value), and the rightmost non-blank value is the one to **survive**.  However, VEDA2 currently does not accept duplicate columns with **fully identical** headers (while VEDA-FE did accept such).